internal/service/advancedclustertpf/resource_compatibility_reuse.go

```diff
@@ -11,6 +12,23 @@ import (
 	"go.mongodb.org/atlas-sdk/v20241113002/admin"
 )
 
+func IsMajorVersionHigherGreaterOrEqual(input *string, version float64) bool {
+	if input == nil || *input == "" {
+		return true
+	}
+	parts := strings.SplitN(*input, ".", 2)
+	if len(parts) == 0 {
+		return false
+	}
+
+	value, err := strconv.ParseFloat(parts[0], 64)
+	if err != nil {
+		return false
+	}
+
+	return value >= version
```

Comment:
arithmetic and equal comparison with floats can be tricky because roundings, i assume it's working fine here